examples/nvram/read.c

```diff
@@ -124,17 +130,42 @@ int TPM2_NVRAM_Read_Example(void* userCtx, int argc, char *argv[])
     nv.handle.auth.size = auth.size;
     XMEMCPY(nv.handle.auth.buffer, auth.buffer, auth.size);
 
-    readSize = keyBlob.pub.size;
-    printf("Trying to read %d bytes of public key part from NV\n", readSize);
-    rc = wolfTPM2_NVReadAuth(&dev, &nv, TPM2_DEMO_NVRAM_STORE_INDEX,
-        (byte*)&keyBlob.pub.publicArea, &readSize, 0);
-    if (rc != 0) goto exit;
+    if (partialRead != PRIVATE_PART_ONLY) {
+        readSize = sizeof(keyBlob.pub);
```

Comment:
Please refactor these to store actual size (first two bytes) and the tpm2_packet encoding/decoding API's, since the NV space on TPM's is limited.

Reply:
Great. I am glad we are willing to spend more time on this to optimise for NVRAM space.